src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,44 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final String SEPARATOR = " ";
+    private final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder finalReport = new StringBuilder();
+        String[] checkLine;
+        boolean isDateInRange;
+        LocalDate dateRangeBegin;
+        LocalDate dateRangeEnd;
+        LocalDate dataSalary;
+
+        finalReport.append("Report for period ")
+                .append(dateFrom).append(" - ")
+                .append(dateTo);
+
+        for (String name : names) {
+            finalReport.append(System.lineSeparator()).append(name).append(" - ");
+            int salary = 0;
+
+            for (String dateString : data) {
+                checkLine = dateString.split(SEPARATOR);
+                dateRangeBegin = LocalDate.parse(dateFrom, formatter);
+                dateRangeEnd = LocalDate.parse(dateTo, formatter);
```

Comment:
Why do we need to reparse these dates for number of names multiplied for number of data rows?
They never changes, we may parse them once during variable initialization

Reply:
sorry, stupid mistake :)))